src/diagnostic-channel-publishers/src/winston.pub.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+import {channel, IModulePatcher, PatchFunction} from "diagnostic-channel";
+
+export interface IWinstonData {
+    message: string;
+    meta: any;
+    level: string;
+}
+
+// register a "filter" with each logger that publishes the data about to be logged
+const winstonPatchFunction: PatchFunction = (originalWinston) => {
+    const originalLog = originalWinston.Logger.prototype.log;
+    const loggingFilter = (level, message, meta) => {
+        channel.publish<IWinstonData>("winston", {level, message, meta});
```

Comment:
Winston allows you to log metadata objects along with string messages, which it then for whatever reason just prints out as space-separated "key=value" strings, rather than even JSONifying it. So, I don't think its particularly useful, but its in the API.